src/scenes/moon/index.ts

```diff
@@ -39,13 +40,76 @@ export class Moon extends Scene {
       }),
       frameRate: 4,
     });
-    this.npcs = this.physics.add.staticGroup();
-    this.npcs.add(this.npc);
-    this.physics.add.collider(this.player, this.npcs);
 
+    const gridEngineConfig = {
+      collisionTilePropertyName: "collides",
+      characters: [
+        {
+          id: "player",
+          sprite: this.player,
+          startPosition: { x: 25, y: 25 },
+          speed: 1,
+        },
+        {
+          id: "singleNpc",
+          sprite: this.npc,
+          startPosition: { x: 24, y: 24 },
+        }
+      ],
+    };
+
+    const npcs: Map<string, any> = new Map<string, any>();
+
+    for (let x = 35; x <= 40; x++) {
+      for (let y = 25; y <= 30; y++) {
+        const spr = this.add.sprite(0, 0, "astronaut");
+        npcs.set(`npc${x}#${y}`, spr);
+
+        gridEngineConfig.characters.push({
+          id: `npc${x}#${y}`,
+          sprite: spr,
+          startPosition: { x, y },
+          speed: 1,
+        });
+      }
+    }
+
+
+    this.gridEngine.create(this.map, gridEngineConfig);
+
+    for (let x = 35; x <= 40; x++) {
+      for (let y = 25; y <= 30; y++) {
+        this.gridEngine.moveRandomly(`npc${x}#${y}`, this.getRandomInt(0, 1500));
+      }
+    }
+
+    this.gridEngine.movementStarted().subscribe(({ charId, direction }: any) => {
+      if (charId == "player") {
+        this.player.anims.play(direction);
+      } else {
+        npcs.get(charId).anims.play(direction);
+      }
+    });
+
+    this.gridEngine.movementStopped().subscribe(({ charId, direction }: any) => {
+      if (charId == "player") {
+        this.player.anims.stop();
+        this.player.anims.play("stay-down");
+      }
```

Comment:
we can consider adding stop animations for other characters as well :) but this is for tests so i wouldnt bother, just something we can consider :)

Reply:
done